compiler/ir/ir.tree/gen/org/jetbrains/kotlin/ir/declarations/IrAnonymousInitializer.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+//This file was generated automatically
+//DO NOT MODIFY IT MANUALLY
+
+package org.jetbrains.kotlin.ir.declarations
+
+import kotlin.Boolean
+import kotlin.Unit
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.ir.ObsoleteDescriptorBasedAPI
+import org.jetbrains.kotlin.ir.expressions.IrBlockBody
+import org.jetbrains.kotlin.ir.symbols.IrAnonymousInitializerSymbol
+import org.jetbrains.kotlin.ir.visitors.IrElementTransformer
+import org.jetbrains.kotlin.ir.visitors.IrElementVisitor
+
+/**
+ * A leaf IR tree element.
+ */
+public abstract class IrAnonymousInitializer : IrDeclarationBase() {
+    @ObsoleteDescriptorBasedAPI
+    public abstract override val descriptor: ClassDescriptor
+
+    public abstract override val symbol: IrAnonymousInitializerSymbol
+
+    public abstract val isStatic: Boolean
+
+    public abstract var body: IrBlockBody
+
+    public override fun <R, D> accept(visitor: IrElementVisitor<R, D>, `data`: D): R =
```

Comment:
> I'd also expect their authors do look at some git diff of the tree structure between compiler versions anyway.

Yes, it's true. But in current approach it's quite hard to understand whats going on as file history is broken. Same thing makes full review complicated. That blocks me (and I assume other folks) to perform it in proper way.
Change for 'visitConst' might be ok (as migration here is quite straightforward and it even keeps binary compatibility:)). 

>At least regarding compiler plugins I was told not to worry too much about them as their still in alpha,

It could be OK in many cases. But here (from my side) core part of IR lowering infrastructure is touched and I would prefer to have some migration mechanism to allow smooth migration. I can be wrong but look like it will require 2-3 additional `if` in code generator.

>But since you want the generated code to be pretty, it should be easier to run some linter over it rather than applying manual corrections, which I don't think would help that much either (I've also dig onto some issue at kotlinpoet and they suggest the same approach).

Most likely you speak about something like code formatter. We used built in one in IDEA and most likely it could be used separately (but it could require some big dependencies that I'm not sure is good to have here). As was already pointed by @demiurg906  IR base declaration assumed to be read in different scenarios and "it's important to have this generated code written in general code style"

Reply:
> Most likely you speak about something like code formatter. 

Yes, in particular the newly announced [qodana](https://www.jetbrains.com/qodana/), though now that I looked it doesn't quite seem to support formatting.
So how about postprocessing with [ktlint](https://ktlint.github.io/)? It would most probably require a few new simple rules for best effect which can be easily plugged-in by creating new module alongside.

Reply:
Yes, same code style is important as was written before, but looks like it's last priority concern that I have //cc @bashor